arviz/utils.py

```diff
@@ -2,25 +2,37 @@
 import importlib
 
 
-def _var_names(var_names):
+def _var_names(var_names, data):
     """Handle var_names input across arviz.
-
     Parameters
     ----------
     var_names: str, list, or None
-
+    data : xarray.Dataset
+        Posterior data in an xarray
     Returns
     -------
     var_name: list or None
     """
-    if var_names is None:
-        return None
-
-    elif isinstance(var_names, str):
-        return [var_names]
-
-    else:
-        return var_names
+    if var_names is not None:
+
+        if isinstance(var_names, str):
+            var_names = [var_names]
+
+        if isinstance(data, (list, tuple)):
+            all_vars = []
+            for dataset in data:
+                dataset_vars = list(dataset.data_vars)
+                for var in dataset_vars:
+                    if var not in all_vars:
+                        all_vars.append(var)
+        else:
+            all_vars = list(data.data_vars)
+
+        exclude_vars = [i[1:] for i in var_names if i.startswith("~")]
```

Comment:
exclude_vars = [i[1:] for i in var_names if i.startswith("~") and i not in all_vars]

Reply:
This will give the wrong result

Reply:
OK, with this change negation will work as expected except that if a variable name starts with '~' like `~mu` when passing `~mu` to `var_names` it will understand that you want the variable `~mu` and not to negate a variable named `mu`. Passing `~~mu` will be understood as negating the variable `~mu`. Is this OK?

Reply:
I think thats fine. We're only promising a negation if the leading character is a tilda

Reply:
But... how do you negate `mu` in that situation?

Reply:
you're saying if a model contains both `mu` and `~mu` how would you negative one but not the other?

Reply:
How do you negate mu. I'm not saying we need to handle it, but warn users

Reply:
For now I think that's reasonable. Scan the var_names in InferenceData and see if any are leading with a tilda. If so provide warning negation behavior might be weird.

If it comes up enough we can solve further?

Reply:
oops, I missed this conversation, I will add the warning.